### PR TITLE
support for parallel by preloading data

### PR DESCRIPTION
**Pull request by Tim-Yu**
Hi,

Related to issue #149 

> is there a particular reason to use a `CompressedGRangesList` instead of a standard `GRAngesList`?

Yes, there are two reasons: 1) to ensure that the data is from cdsBy() etc.; 2) for more memory efficiency.

I have updated the remaining coordinate mapping functions to support parallel process calls, along with the corresponding unit tests. 

Additionally, I noticed a minor typo in 2db48b6f28a6526916decfa07145484b6659d1ed and a function error in e5f41605b630a47fad601ce41b4b5edf16d84e8b. 

Best,

Boyu

### CODE REVIEW

File: R/proteinToX.R [hunk 1]
```diff
@@ -100,8 +104,11 @@
 #'
 #' ## The result for the region within the second protein
 #' res[[2]]
-proteinToTranscript <- function(x, db, id = "name",
-                                idType = "protein_id") {
+setGeneric("proteinToTranscript", signature="db",
```
Comment (jorainer) [inline R/proteinToX.R:109]: Please make sure to also document the `...`

File: R/genomeToX.R [hunk 2]
```diff
@@ -75,11 +76,40 @@
...
+#'     filter = AnnotationFilterList(
+#'         SeqNameFilter(as.character(unique(seqnames(gnm)))),
+#'         GeneStartFilter(max(end(gnm)), condition = "<="),
+#'         GeneEndFilter(min(start(gnm)), condition = ">=")
+#'     )
+#' )
+#' 
+#' res_temp <- mclapply(1:10000, function(ind){
+#'     genomeToTranscript(gnm[ind], exons)
+#' }, mc.preschedule = T, mc.cores = detectCores() - 1)
```
Comment (jorainer) [inline R/genomeToX.R]: Please use `TRUE` instead of `T`.

**Review by jorainer — CHANGES_REQUESTED** [review]
Thanks for the (quite extensive) contribution! Please address the two small requests to ensure the unit tests run through. And then please also add yourself as an author to the DESCRIPTION.

File: R/genomeToX.R (hunk 2 above)
Comment (Tim-Yu) [reply R/genomeToX.R]: I changed to parLappy for passing the windows check

**Review by jorainer — APPROVED** [review]
Thanks @Tim-Yu ! I'll merge now and push to BioC.

**Pull request merged**